flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/cursor/stop/MessageIdStopCursor.java

```diff
@@ -43,7 +42,7 @@ public MessageIdStopCursor(MessageId messageId, boolean inclusive) {
         checkArgument(!earliest.equals(messageId), "MessageId.earliest is not supported.");
         checkArgument(!latest.equals(messageId), "Use LatestMessageStopCursor instead.");
 
-        this.messageId = convertToMessageIdImpl(messageId);
+        this.messageId = messageId;
```

Comment:
I prefer to use the `MessageIdImpl` here. Pulsar introduced a lot of implementation of `MessageId`. We shall use this conversion to check if it was valid for avoiding some user implemented (if it exists) message ids.

Reply:
Can you add a code suggestion directly? This interface is changed a lot by @bewaremypower's patch - 

https://github.com/apache/pulsar/commit/8c50a6c2e91c81dbf187ce5e66cb39e2758a741e

The inheritance tree of `MessageId` and `MesssageIdImpl` changed.